pennylane/drawer/tape_text.py

```diff
@@ -41,8 +41,15 @@ def _add_op(op, layer_str, wire_map, decimals, cache):
     layer_str = _add_grouping_symbols(op, layer_str, wire_map)
 
     control_wires = op.control_wires
-    for w in control_wires:
-        layer_str[wire_map[w]] += "C"
+    control_values = op.hyperparameters.get("control_values", None)
+    for i, w in enumerate(control_wires):
```

Comment:
You could instead use a `zip` here instead:
```suggestion
    for w, val in zip(control_wires, control_values):
```
This would require moving the `if control_values` / `if 'control_values'  in op.hyperparameters` check outside of the for-loop.

It's always good to perform these kinds of checks once, instead of having to perform the same check each time in a loop.

This code block would also be a good candidate for the single-line if statement syntax:
```
layer_str[wire_map[w]] = "C" is val == "1" else "O"
```

Reply:
I have implemented this